File: aspect/src/main/java/br/com/leonardoferreira/poc/controller/ContactController.java
package br.com.leonardoferreira.poc.controller;

import br.com.leonardoferreira.poc.domain.Contact;
import br.com.leonardoferreira.poc.service.ContactService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/contacts")
public class ContactController {
    @Autowired
    private ContactService contactService;

    @GetMapping
    public List<Contact> findAll(@ModelAttribute Contact contact) {
        return contactService.findAll(contact);
    }

    @PostMapping
    public HttpEntity<?> create(@RequestBody Contact contact) {
        contactService.create(contact);
        return new ResponseEntity(HttpStatus.CREATED);
    }
}

File: postgres-copy/src/main/kotlin/br/com/leonardoferreira/postgres/copy/controller/ContactController.kt
package br.com.leonardoferreira.postgres.copy.controller

import br.com.leonardoferreira.postgres.copy.extension.runWithStopWatch
import br.com.leonardoferreira.postgres.copy.service.BulkContactService
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.multipart.MultipartFile

@RestController
@RequestMapping("/contacts")
class ContactController(
    private val bulkContactService: BulkContactService
) {

    private val log = LoggerFactory.getLogger(ContactController::class.java)

    @PostMapping
    fun bulkInsert(
        @RequestParam("file") multipartFile: MultipartFile
    ): ResponseEntity<*> {
        runWithStopWatch {
            bulkContactService.bulkInsert(multipartFile)
        }
        
        return ResponseEntity<Void>(HttpStatus.CREATED)
    }

}
